src/Zip.Shared/ZipEntry.Write.cs

```diff
@@ -248,11 +248,24 @@ Disk Start
             }
             else
             {
-                // If reading a segmneted archive and saving to a regular archive,
-                // ZipEntry._diskNumber will be non-zero but it should be saved as
-                // zero.
-                bytes[i++] = 0;
-                bytes[i++] = 0;
+                if (_presumeZip64)
+                {
+                    // the Extra block for Zip64 always includes the 'Disk Start',
+                    // so we MUST also set them to -1 (0xFFFF) and direct readers
+                    // to the Zip64 field. Not doing this would display warnings
+                    // in 7-Zip and fail to extract with older versions of
+                    // System.IO.Compression. (GitHub issue #260)
+                    bytes[i++] = 0xFF;
+                    bytes[i++] = 0xFF;
+                }
+                else
+                {
+                    // If reading a segmneted archive and saving to a regular archive,
```

Comment:
```suggestion
                    // If reading a segmented archive and saving to a regular archive,
```

Reply:
I specifically did _not_ fix this typo because it was existing code that I just indented.

I'll let a maintainer decide if they want it though.